tasks/deploy.yml

```diff
@@ -0,0 +1,33 @@
+---
+
+- block:
+
+  - name: Create folders to synchronize
+    file:
+      path: /opt/mattermost-versions/{{ mattermost_version }}/{{ item }}
+      state: directory
+      recurse: yes
+    with_items: '{{ mattermost_preserve_on_update }}'
+
+  # We stop the service before synchronizing such that no data is changed after synchronization
+  - name: Stop mattermost service
+    service:
+      name: mattermost
+      enabled: yes
+      state: stopped
+
+  - name: Synchronize the preserved folders
+    command: rsync -aI /opt/mattermost/{{ item }}/ /opt/mattermost-versions/{{ mattermost_version }}/{{ item }}
+    with_items: '{{ mattermost_preserve_on_update }}'
+
+  when: mattermost_previous_version is defined
+
+- name: Set the symlink to the installed mattermost version
+  file:
+    src: /opt/mattermost-versions/{{ mattermost_version }}
+    dest: /tmp/mattermost
```

Comment:
Is this step really necessary? Why not override the original symlink, shouldn't that also be atomic?
If you have a reason to keep it this way, please rename the symlink so it won't be confused with the extracted mattermost archive in `update.yml`.

Reply:
No, changing a symlink involvis the system calls
  * unlink
  * symlink
See: http://blog.moertel.com/posts/2005-08-22-how-to-change-symlinks-atomically.html

2. Ok.